File: android/app/src/main/java/com/frameprocessortest/dumb/DumbFrameProcessorPlugin.java
package com.frameprocessortest.dumb;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.mrousavy.camera.frameprocessor.Frame;
import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin;

import java.util.HashMap;
import java.util.Map;

public class DumbFrameProcessorPlugin extends FrameProcessorPlugin {
  DumbFrameProcessorPlugin(@Nullable Map<String, Object> options) {
    super(options);
  }

  @Nullable
  @Override
  public Object callback(@NonNull Frame frame, @Nullable Map<String, Object> arguments) {
    Map<String, Object> result = new HashMap<>();
    result.put("dumb", "result");
    return result;
  }
}

File: ios/frameProcessorTest/DumbFrameProcessorPlugin.swift
import VisionCamera

@objc(DumbFrameProcessorPlugin)
public class DumbFrameProcessorPlugin: FrameProcessorPlugin {
  public override init(options: [AnyHashable: Any]! = [:]) {
    super.init(options: options)
  }

  public override func callback(_ frame: Frame, withArguments arguments: [AnyHashable: Any]?) -> Any? {
    return [
      "dumb": "result"
    ]
  }
}
